plugins/field-angle/src/field_angle.ts

```diff
@@ -0,0 +1,553 @@
+   * These values need to be rounded and wrapped before being displayed so
+   * that the text input's value is appropriate.
+   *
+   * @param angle New angle.
+   */
+  private displayMouseOrKeyboardValue(angle: number) {
+    if (this.round) {
+      angle = Math.round(angle / this.round) * this.round;
+    }
+    angle = this.wrapValue(angle);
+    if (angle !== this.value_) {
+      this.setEditorValue_(angle);
+    }
+  }
+
+  /** Redraw the graph with the current angle. */
+  private updateGraph() {
+    if (!this.gauge || !this.line) {
+      return;
+    }
+    // Always display the input (i.e. getText) even if it is invalid.
+    let angleDegrees = Number(this.getText()) + this.offset;
+    angleDegrees %= 360;
+    let angleRadians = Blockly.utils.math.toRadians(angleDegrees);
+    let path = `M ${FieldAngle.HALF},${FieldAngle.HALF}`;
+    let x2 = FieldAngle.HALF;
+    let y2 = FieldAngle.HALF;
+    if (!isNaN(angleRadians)) {
+      const clockwiseFlag = Number(this.clockwise);
+      const angle1 = Blockly.utils.math.toRadians(this.offset);
+      const x1 = Math.cos(angle1) * FieldAngle.RADIUS;
+      const y1 = Math.sin(angle1) * -FieldAngle.RADIUS;
+      if (clockwiseFlag) {
+        angleRadians = 2 * angle1 - angleRadians;
+      }
+      x2 += Math.cos(angleRadians) * FieldAngle.RADIUS;
+      y2 -= Math.sin(angleRadians) * FieldAngle.RADIUS;
+      // Don't ask how the flag calculations work.  They just do.
+      let largeFlag =
+          Math.abs(Math.floor((angleRadians - angle1) / Math.PI) % 2);
+      if (clockwiseFlag) {
+        largeFlag = 1 - largeFlag;
+      }
+      path += ` l ${x1},${y1} A ${FieldAngle.RADIUS},${FieldAngle.RADIUS} 0 ` +
+          `${largeFlag} ${clockwiseFlag} ${x2},${y2} z`;
+    }
+    this.gauge.setAttribute('d', path);
+    this.line.setAttribute('x2', `${x2}`);
+    this.line.setAttribute('y2', `${y2}`);
+  }
+
+  /**
+   * Handle key down to the editor.
+   *
+   * @param e Keyboard event.
+   */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  protected override onHtmlInputKeyDown_(e: KeyboardEvent) {
+    super.onHtmlInputKeyDown_(e);
+    const block = this.getSourceBlock();
+    if (!block) {
+      throw new Error('The field has not yet been attached to its input. ' +
+          'Call appendField to attach it.');
+    }
+
+    let multiplier = 0;
+    switch (e.key) {
+      case 'ArrowLeft':
+        // decrement (increment in RTL)
+        multiplier = block.RTL ? 1 : -1;
+        break;
+      case 'ArrowRight':
+        // increment (decrement in RTL)
+        multiplier = block.RTL ? -1 : 1;
+        break;
+      case 'ArrowDown':
+        // decrement
+        multiplier = -1;
+        break;
+      case 'ArrowUp':
+        // increment
+        multiplier = 1;
+        break;
+    }
+    if (multiplier) {
+      const value = this.getValue() as number;
+      this.displayMouseOrKeyboardValue(value + multiplier * this.round);
+      e.preventDefault();
```

Comment:
Why only call `preventDefault()` and `stopPropagation()` when there's a multiplier?

Edit: ah, I see, because otherwise it's another key and you ignore it. How about putting a default case in the `switch` that returns early, and then doing everything in the `if (multiplier)` section without the conditional?